### ISSUE TITLE
Track availability / installation of Intel oneAPI (LLVM) compilers on tier1 hosts

### ISSUE THREAD

**Issue by rickgrubin-noaa** [JCSDA/spack-stack#1325]
**Is your feature request related to a problem? Please describe.**

Track the availability, and if necessary, `spack-stack` installation, of `Intel oneAPI `(`LLVM`) compilers on **`tier1`** hosts.

**Describe the solution you'd like**

Note: require `oneapi@2024.2.1` or later because of [current support](https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-supported-product-versions.html) of, and bugs in, previous releases.

Current status of each [tier1](https://github.com/JCSDA/spack-stack/tree/develop/configs/sites/tier1) host is listed below, per [release/1.8.0](https://github.com/JCSDA/spack-stack/tree/release/1.8.0) `configs/sites/tier1/<host>/compilers.yaml` or if _`host`_ is marked with an asterisk (**_*_**), verified via `module available` on the _`host`_ itself. This is also shown below for each host.

**NB**: 
- do not have access to all `tier1` hosts, please update the list if able.

**Hosts**

- `acorn`
  - [ ] `intel@19.1.3.304` (per `configs/sites/tier1/acorn/compilers.yaml`)
- `atlantis`
  - [x] `oneapi@2025.0.3` and more (per `configs/sites/tier1/atlantis/compilers.yaml`)
- `derecho *` 
  - [x] 2024.2.1 (via `module available`)
- `discover`
  - `scu-17`
    - [ ] `intel@2021.10.0` (per `configs/sites/tier1/discover-scu17/compilers.yaml`)
- `gaea *`
  - `c5 *`
    - [x] `intel@2024.2.1` 
    - installed via `spack-stack`
    - `module use module use /ncrc/proj/epic/spack-stack/modulefiles`
    - `module load intel-oneapi-compilers/2024.2.1`
  - `c6 *`
    - [x] `intel@2024.2.1` 
    - installed via `spack-stack`
    - `module use module use /ncrc/proj/epic/spack-stack/c6/modulefiles`
    - `module load intel-oneapi-compilers/2024.2.1`

Via email from **GFDL** 21-October-2-24:

> Update:
> Currently, the CPE does not yet support 2024.2 -- we are waiting on word from ORNL on when this compiler will be supported.
> 
> For now, users can still use 2024.2 on c5/c6 at their own risk. cc and CC wrappers are not expected to work.
> 
> In addition, intel packages are not managed in a sensical manner. When there is an update to Intel 2024.2 release (i.e. from 2024.2.0 -> 2024.2.1 ) they put everything into the 2024.2 package repo location instead of making a separate release for 2024.2.1. ORNL does not have much control if what will be available in the 2024.2 module will be exactly what the user had wanted from the specific 2024.2.1 release, nor can they specifically install 2024.2.1 version because Intel doesn't do version and releases properly. 

- `hera *`
  - [x] `intel@2024.2.1` (via `module available`)
- `hercules *`
  - [x] `intel@2024.2.1`  installed via `spack-stack`
  - installed via `spack-stack`
  - `module use /work/noaa/epic/role-epic/spack-stack/hercules/modulefiles`
  - `module load intel-oneapi-compilers/2024.2.1`
  - [x] `intel@2024.2.1` installed as a system module
  - `module load spack-managed-x86-64_v3`
  - `module load intel-oneapi-compilers/2024.2.1`
- `jet *`
  - [x] `oneapi@2024.2.1` (via`module available`)
- `narwhal`
  - [x] `oneapi@2024.2.1` and others (per `configs/sites/tier1/narwhal/compilers.yaml`)
- `nautilus`
  - [x] `oneapi@2024.2.1` (per `configs/sites/tier1/nautilus/compilers.yaml`
- `orion`
  - [x] `intel@2024.2.1`
  - installed via `spack-stack`
  - `module use /work/noaa/epic/role-epic/spack-stack/orion/modulefiles`
  - `module load intel-oneapi-compilers/2024.2.1`
  - [x] `intel@2024.2.1` installed as a system module
  - `module load spack-managed-x86-64_v3`
  - `module load intel-oneapi-compilers/2024.2.1`
- `s4`
  - [ ] `intel@2021.5.0` (per `configs/sites/tier1/s4/compilers.yaml`)

- `CSP`
  - `noaa-aws *`
    - [ ] `intel@2021.3.0`
  - `noaa-azure *`
    - [ ] `intel@2021.3.0`
  - `noaa-gcloud *`
    - [ ] `intel@2021.3.0`

- `ursa`
  - [x] `oneapi@2024.2.1` (via`module available`)
    - see also [Add new machine: NOAA Ursa (Hera's follow-on)](https://github.com/JCSDA/spack-stack/issues/1297)

**Additional context**

Requests to upgrade to the latest supported version of `oneAPI` compilers (see link below) have been made on all NOAA tier1 hosts.

[Intel® Toolkits, AI Tools, Compilers Supported Product Versions](https://www.intel.com/content/www/us/en/developer/articles/release-notes/oneapi-supported-product-versions.html)

**Comment by rickgrubin-tomorrow** [comment]
Regarding progress on **_hera_**, see: [Install spack-stack 1.8.0 (and all add-on envs needed for UFS WM @ develop) with LLVM compilers on Hera #1345](https://github.com/JCSDA/spack-stack/issues/1345)

**Comment by climbfuji** [comment]
I think this is quite outdated and no longer needed. Do we have oneapi compilers on all systems nowadays? I believe the answer is yes.
